core/Piwik.php

```diff
@@ -262,6 +263,51 @@ public static function checkUserHasSuperUserAccessOrIsTheUser($theUser)
         }
     }
 
+    /**
+     * Request a token auth to authenticate in a request.
+     *
+     * Note: During one request the token is only being requested once and used throughout the request. So you want to make
+     * sure the token is valid for enough time for the whole request to finish.
+     *
+     * @param string $reason some short string/text explaining the reason for the token generation, eg "CliMultiAsyncHttpArchiving"
+     * @param int $validForHours For how many hours the token should be valid. Should not be valid for more than 14 days.
+     * @return mixed
+     */
+    public static function requestTemporarySystemAuthToken($reason, $validForHours)
+    {
+        static $token = array();
+
+        if (isset($token[$reason])) {
+            // note: For now we do not increase the expire time when it is already requested
+            return $token[$reason];
+        }
+
+        $twoWeeksInHours = 14 * 24;
+        if ($validForHours > $twoWeeksInHours) {
+            throw new Exception('The token cannot be valid for so many hours: ' . $validForHours);
+        }
```

Comment:
Can we just set `$validForHours` to `$twoWeeksInHours` in this case? Might avoid some fatal errors in the future. (I can't imagine a case where someone would do this on purpose in a plugin, so I assume it would always be a strange bug, if it occurs at all.)

Reply:
Not sure @diosmosis personally, would prefer to throw an exception. It be only mostly under development anyway if someone calls it wrongly. Otherwise they might pause 4 weeks and don't notice it won't work because it's changed silently which might not be expected.

I could throw the exception though if needed only if Dev Mode is enabled if that helps? And fallback to 2 weeks if dev mode is disabled but then it might be hard to debug some issues.

Reply:
We can always remove the exception later if it becomes an issue, so probably not an issue.